incubator/blockchain-client-openliberty/templates/default/README.md

```diff
@@ -0,0 +1,96 @@
+
+### Identity Request Flow
+
+1. IdentityMapperFilter intercepts the request ahead of being routed to the API resource.  (Pre request filter)
+1. Invokes extractPrincipal(ContainerRequestContext) to get the request identity.
+1. By default the filter method then returns the default identity, based on the FABRIC_DEFAULT_IDENTITY environment variable.
+In a more sophisticated implementation, the mapper would call out to a rule engine, or to a database to perform the mapping.
+1. The identity is then set in an HTTP Header named X-FABRIC-IDENTITY.
+1. The request now gets routed to the API resource.
+1. The resource logic extracts the identity from the X-FABRIC-IDENTITY header and uses it to call getContract(identity) on the ConnectionManager.
+
+## Required Envirnment Variables
+
+Here is a complete list of the supported environment variables as well as an example configuration.
+
+***FABRIC_CONNECTION_PROFILE*** : Blockchain standard document that determines how a gateway connects to a fabric network.  <https://hyperledger-fabric.readthedocs.io/en/release-2.0/developapps/connectionprofile.html> This document should be "compressed/minified" so that there are no spaces or line breaks.
+
+***FABRIC_WALLET_PROFILE*** : Setting to specify how the service finds and loads the identies. There are two supported types out of the box. In memory or filesystem. This document should be "compressed/minified" so that there are no spaces or line breaks.
+
+`{"type":"IN_MEMORY"}`
+
+`{"type":"FILE_SYSTEM","options":{"path":"/project/user-app/src/main/java/application/Org1"}}`
+
+option.path : is that path to the file system wallet structure in compliance with the SDK wallet structure. This file MUST be included in the project and is for ***DEVELOPMENT PURPOSES ONLY***.
+
+***FABRIC_WALLET_CREDENTIALS*** : A JSON array of fabric identies used when using IN_MEMORY wallet type. (See sample below). This document should be "compressed/minified" so that there are no spaces or line breaks. Additionally, the cert and private_key attribute values ***MUST be base64 encoded.***
+
+***FABRIC_CHANNEL*** : Fabric network channel
+***FABRIC_CONTRACT*** : Installed and Instantiated Smart Contract on the FABRIC_CHANNEL
+***FABRIC_DEFAULT_IDENTITY*** : The default identity to use when transacting with the fabric network.
```

Comment:
There is a correlation here between the content of FABRIC_WALLET_CREDENTIALS and also the wallet we are using.  We should be clear about this. Basically in a more complex environment where API callers are mapped to Fabric identities there is really no need for this default identity.   In that scenario an authenticated API caller will map to a fabric identity and the IdentityMapper component would do that for you.  In the case of this basic example since the API is not being authenticated we use this FABRIC_DEFAULT_IDENTITY to tell us what should be the name of the identity we should use by default.   This value should map to the name field of one of the identities injected through FABRIC_WALLET_CREDENTIALS or the name field of one of the identities found in the filesystem when using the Filesystem Wallet.